test/SILGen/wildcard_dispatch_on_catch.swift

```diff
@@ -0,0 +1,26 @@
+// RUN: $target-swift-frontend -emit-silgen -verify %s
+
+enum testError: ErrorProtocol {
+   case C1
+   case C2(Int, String)
+}
+
+public func someFunc(_ str: String) throws -> String {
+   return str
+}
+
+let testData: [String] = [
+   "ABC",
+   "DEF",
+   ]
+
+public func testCatchWildcardDispatch(_ name: String...) throws {
```

Comment:
Since you're not using FileCheck to verify the SILGen output in this test, it is better to move it to tests/Interpreter/ and make it an executable test, so you can exercise IRGen as well.